middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -773,6 +773,7 @@ type simplified_block_load =
   | Not_a_block
   | Block_but_cannot_simplify of Code_or_metadata.t Value_approximation.t
   | Field_contents of Symbol.t
+  | Tagged_immediate of Targetint_31_63.t
```

Comment:
Note: after reading the code again, I think it might be simpler to fold this into the `Field_contents` case, using `Simple.t` as the payload instead of `Symbol.t`. Or alternatively, renaming `Field_contents` to something more explicit like `Field_as_symbol`.

Reply:
Agreed. It might also be a bit more future-proof. I'll see to it.